src/main/java/com/mycompany/thevoid/EnemyName.java

```diff
@@ -89,65 +87,46 @@ public String setName() {
         returnedName += secondString;
         String thirdString = selectName(lastName, weightLast, true);
         returnedName += thirdString;
-        
-        
-        
 
         return returnedName;
     }
-    
-    public boolean isName(int chance){
+
+    public boolean isName(int chance) {
         int randomNumber = (int) (Math.random() * (101 - 1)) + 1;
-        
-        if (randomNumber < chance)
-            return true;
-        else return false;
+        return randomNumber < chance;
     }
-    
-    
-    public String selectName(ArrayList names, ArrayList weight, boolean isLastName){ //firstName firstWeight
+
+    public String selectName(ArrayList<String> names, ArrayList<Integer> weight, boolean isLastName) { //firstName firstWeight
```

Comment:
Oh, so that's why I was unable to pull weight directly from parameters. Thanks for the insight!

Reply:
Yes because by default without type, it will be ArrayList<Objects> then we have to manually type cast them.